### PR TITLE
fix: prevent click event when publish button is disabled

### LINKED ISSUE #1480: prevent click event when pubish button is disabled
CPT2301271417-639x277
It looks weird now. If publish button is disabled, it should not trigger any event or network request.

### PR DESCRIPTION
**Pull request by sechi747**
closes #1480

### CODE REVIEW

**Comment by ushuz** [comment]
Maybe add `disabled="isPublishedDisabled"` to the publish button as well?

https://github.com/elk-zone/elk/blob/1b4202d6a6994a1ab5b7319c46078bb22ebe12a0/components/publish/PublishWidget.vue#L324-L331

**Comment by sechi747** [comment]
> 

I try it, but it'll make tooltip disappear and the cursor will change to `default`

File: components/publish/PublishWidget.vue [hunk 1]
```diff
@@ -97,6 +97,8 @@ async function toggleSensitive() {
 }
 
 async function publish() {
+  if (isPublishDisabled)
```
Comment (userquin) [inline components/publish/PublishWidget.vue:100]: can you move this check to `publishDraft` on `publish.ts` module'? rn only used here...

Comment (sechi747) [reply components/publish/PublishWidget.vue]: > can you move this check to `publishDraft` on `publish.ts` module? rn only used here...

sure, I have resolved it.

**Pull request merged**